### DISCUSSION TITLE
Design Pattern vs Design Principles

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#140] [Q&A] [answered]
Hello,

What are the differences between Design Patterns and Design Principles?

Is DI(dependency injection) a design pattern or a design principle? 

Thank you

**Comment by christiannagel** [comment, marked as answer]
[Varun](https://stackoverflow.com/users/3454208/varun) gives a convenient answer here: https://stackoverflow.com/questions/31317141/whats-the-difference-between-design-patterns-and-design-principles

A **design principle** provides high-level guidelines without implementations, e.g. *Single Responsibility Principle*, *Open/Closed Principle*, *Liskov Substitution Principle*, *Interface Segregation Principle*, *Dependency Inversion Principle*.

A **design pattern** suggests a specific implementation for a specific object-oriented programming problem.

You can also check the Wikipedia article [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection) describing the **dependency injection design pattern**.
